src/main/java/decodes/cwms/algo/ResEvapAlgo.java

```diff
@@ -329,111 +400,142 @@ public double getMaxTempDepthMeters(){
-//			is_english = true;
-//		}
-
-		double[] elev_array = new double[rcount];
-		double[] area_array = new double[rcount];
-		for ( int i =0; i< rcount; i++){
-			RatingValue rv = rt[i];
-			elev_array[i] = rv.getIndValue();
-			area_array[i] = rv.getDepValue();
-		}
-		reservoir.setElevAreaCurve(elev_array, area_array, rcount, is_english);
+		tally = 0.0;
+		count = 0;
+		if (baseTimes.size() == 24) {
+
+			//initialize database connections
+			siteDAO = tsdb.makeSiteDAO();
+			timeSeriesDAO = tsdb.makeTimeSeriesDAO();
+			crd = new CwmsRatingDao((CwmsTimeSeriesDb) tsdb);
+
+			//Get site Data from Database
+			try {
+				DbKey siteID = siteDAO.lookupSiteID(reservoirId);
+				site = siteDAO.getSiteById(siteID);
+			} catch (DbIoException | NoSuchObjectException e) {
+				throw new RuntimeException("Failed to load Site data", e);
+			}
+
+			//If missing data overwrite with site info
+			if(Longi==0){
+				Longi = Double.parseDouble(site.longitude);
+			}
+			if(Lati==0){
+				Lati = Double.parseDouble(site.latitude);
+			}
+			if(isMissing(Timezone)){
+				Timezone = site.timeZoneAbbr;
+			}
 
+			//initialized Water Temperature Profiles
+			hourlyWTP = new WaterTempProfiles(timeSeriesDAO, start_depth, depth_increment);
+			DailyWTP = new WaterTempProfiles(timeSeriesDAO, start_depth, depth_increment);
+
+			//initialized input timeseries
+			HourlyEvapTS = getParmRef("HourlyEvap").timeSeries;
+			DailyEvapTS = getParmRef("DailyEvap").timeSeries;
+
+			windSpeedTS = getParmRef("windSpeed").timeSeries;
+			AirTempTS = getParmRef("AirTemp").timeSeries;
+			RelativeHumidityTS = getParmRef("RelativeHumidity").timeSeries;
+			AtmPressTS = getParmRef("AtmPress").timeSeries;
+			PercentLowCloudTS = getParmRef("PercentLowCloud").timeSeries;
+			ElevLowCloudTS = getParmRef("ElevLowCloud").timeSeries;
+			PercentMidCloudTS = getParmRef("PercentMidCloud").timeSeries;
+			ElevMidCloudTS = getParmRef("ElevMidCloud").timeSeries;
+			PercentHighCloudTS = getParmRef("PercentHighCloud").timeSeries;
+			ElevHighCloudTS = getParmRef("ElevHighCloud").timeSeries;
+			ElevTS = getParmRef("Elev").timeSeries;
+
+
+			//initialize MetData
+			EvapMetData metData = new EvapMetData();
+			metData.setWindSpeedTs(windSpeedTS);
+			metData.setAirTempTs(AirTempTS);
+			metData.setRelHumidityTs(RelativeHumidityTS);
+			metData.setAirPressureTs(AtmPressTS);
+			metData.setLowCloudTs(PercentLowCloudTS, ElevLowCloudTS);
+			metData.setMedCloudTs(PercentMidCloudTS, ElevMidCloudTS);
+			metData.setHighCloudTs(PercentHighCloudTS, ElevHighCloudTS);
+
+			//initialize Evaporation Reservoir
+			reservoir = new EvapReservoir();
+			reservoir.setName(reservoirId);
+			reservoir.setThermalDiffusivityCoefficient(ThermalDifCoe);
+			try {
+				reservoir.setWindShearMethod(WindShearMethod.fromString(WindShear));
+			} catch (Throwable ex) {
+				ex.printStackTrace();
+			}
 
+			reservoir.setInputDataIsEnglish(true);
+			double lonneg = -Longi; // why make longitude positive?
```

Comment:
Are we "correcting" user data here? In theory a negative Longitude could be correct for a future OpenDCS user.